src/state/selectors.ts

```diff
@@ -86,6 +88,7 @@ export const selectAssetIdParamFromFilter = selectParamFromFilter('assetId')
 export const selectChainIdParamFromFilter = selectParamFromFilter('chainId')
 export const selectUserStakingIdParamFromFilter = selectParamFromFilter('userStakingId')
 export const selectStakingIdParamFromFilter = selectParamFromFilter('stakingId')
+export const selectOpportunityIdParamFromFilter = selectParamFromFilter('opportunityId')
```

Comment:
preferably-blocking: revert me and use `stakingId` param selector above since we know this is a `StakingId`, this isn't used anywhere else

Reply:
Or actually we can probably revert me and not even use `selectEarnUserStakingOpportunityByOpportunityId` altogether, see comment below

Reply:
It's actually needed in the case you don't have a wallet connected, you wan't to pick a parsed opportunity with zeroed values, but in case you have a connected wallet, you wan't to pick the actual existing opportunity

We could remove this selector and add dummy datas directly in the component, it might be cleaner? The actual intent from this new selector is to return an actual parsed opportunity in any cases to display something zeroed out at least

Reply:
Intent sounds sane to me! Though current app logic should be able to handle that just fine?
i.e:

- Staking opportunity can be selected with `selectStakingOpportunityByFilter` 
- *user* staking opportunity can be selected with `selectUserStakingOpportunityByUserStakingId` 

Alternatively, if we're confident zeroing out values works across the board, we could use the current `selectEarnUserStakingOpportunityByUserStakingId` and rename it to `selectEarnUserStakingOpportunityByFilter` with both `userStakingId` current filter (`selectUserStakingOpportunityByUserStakingId`) and `stakingId` new filter (`selectStakingOpportunityByFilter`) and rely on the opportunity without userdata and zero'd out values in case no user staking opportunity is found?

Reply:
Done in [527f86c](https://github.com/shapeshift/web/pull/7934/commits/527f86ce55071bc81f46808931a053d3e5ee04e3)